packages/electron-builder-squirrel-windows/src/SquirrelWindowsTarget.ts

```diff
@@ -20,31 +22,57 @@ export default class SquirrelWindowsTarget extends Target {
 
-    const files = await fs.promises.readdir(tmpVendorDirectory)
-    for (const file of files) {
-      if (["Squirrel.exe", "StubExecutable.exe"].includes(file)) {
-        const filePath = path.join(tmpVendorDirectory, file)
-        log.debug({ file: filePath }, "signing vendor executable")
-        await this.packager.sign(filePath)
-      }
+    if (isEmptyOrSpaces(customSquirrelVendorDirectory) || !fs.existsSync(customSquirrelVendorDirectory)) {
+      log.warn({ customSquirrelVendorDirectory: customSquirrelVendorDirectory }, "unable to access custom Squirrel.Windows vendor directory, falling back to default vendor ")
+      const windowInstallerPackage = require.resolve("electron-winstaller/package.json")
+      const vendorDirectory = path.join(path.dirname(windowInstallerPackage), "vendor")
+
+      const squirrelBin = await getBin(
+        "squirrel.windows",
+        "https://github.com/electron-userland/electron-builder-binaries/releases/download/squirrel.windows@1.0.0/squirrel.windows-2.0.1-patched.7z",
+        "DWijIRRElidu/Rq0yegAKqo2g6aVJUPvcRyvkzUoBPbRasIk61P6xY2fBMdXw6wT17md7NzrTI9/zA1wT9vEqg=="
+      )
+
+      await fs.promises.cp(vendorDirectory, tmpVendorDirectory, { recursive: true })
+      // copy the patched squirrel to tmp vendor directory
+      await fs.promises.cp(path.join(squirrelBin, "electron-winstaller", "vendor"), tmpVendorDirectory, { recursive: true })
+    } else {
+      // copy the custom squirrel vendor directory to tmp vendor directory
+      await fs.promises.cp(customSquirrelVendorDirectory, tmpVendorDirectory, { recursive: true })
     }
 
+    const files = await fs.promises.readdir(tmpVendorDirectory)
+    const squirrelExe = files.find(f => f === "Squirrel.exe")
+    if (squirrelExe) {
+      const filePath = path.join(tmpVendorDirectory, squirrelExe)
+      log.debug({ file: filePath }, "signing vendor executable")
+      await this.packager.sign(filePath)
+    } else {
+      log.warn("Squirrel.exe not found in vendor directory, skipping signing")
+    }
     return tmpVendorDirectory
   }
 
+  private async generateStubExecutableExe(appOutDir: string, vendorDir: string) {
+    const files = await fs.promises.readdir(appOutDir, { withFileTypes: true })
+    const appExe = files.find(f => f.name === `${this.appName}.exe`)
+    if (appExe) {
+      const filePath = path.join(appOutDir, appExe.name)
+      log.filePath(filePath)
+      const stubExePath = path.join(appOutDir, `${this.appName}_ExecutionStub.exe`)
+      await fs.promises.copyFile(path.join(vendorDir, "StubExecutable.exe"), stubExePath)
+      await execWine(path.join(vendorDir, "WriteZipToSetup.exe"), null, ["--copy-stub-resources", filePath, stubExePath])
+      await this.packager.sign(stubExePath)
+      log.debug({ file: filePath }, "signing app executable")
+      await this.packager.sign(filePath)
+    } else {
+      log.warn("App executable not found in app directory, skipping signing")
```

Comment:
Wait, should this edge case ever exist or would it be a broken installer/app? Wondering if we should just
```
if (!appExe) {
   throw new Error("App executable not found in app directory")
}
```